test/unit/org/apache/cassandra/repair/autorepair/FixedSplitTokenRangeSplitterHelper.java

```diff
@@ -127,39 +101,30 @@ public void testTokenRangesSplitByTable()
         assertFalse(keyspaceAssignments.hasNext());
 
         List<RepairAssignment> assignments = keyspace.getRepairAssignments();
-        assertEquals(totalTokenRanges*numberOfSplits*tables.size(), assignments.size());
+        assertEquals(numTokens * numberOfSplits * tables.size(), assignments.size());
         assertEquals(expectedToken.size(), assignments.size());
 
-        int expectedTableIndex = -1;
-        for (int i = 0; i < totalTokenRanges * numberOfSplits * tables.size(); i++)
-        {
-            if (i % (totalTokenRanges * numberOfSplits) == 0)
-            {
-                expectedTableIndex++;
-            }
-        }
-
-        expectedTableIndex = -1;
-        // should be a set of ranges for each table.
-        for (int i = 0; i<totalTokenRanges*numberOfSplits*tables.size(); i++)
+        int assignmentsPerTable = numTokens * numberOfSplits;
+        for (int i = 0; i < tables.size(); i++)
         {
-            if (i % (totalTokenRanges*numberOfSplits) == 0)
+            List<RepairAssignment> assignmentForATable = new ArrayList<>();
+            List<Range<Token>> expectedTokensForATable = new ArrayList<>();
+            for (int j = 0; j < assignmentsPerTable; j++)
             {
-                expectedTableIndex++;
+                assertEquals(Arrays.asList(tables.get(i)), assignments.get(i * assignmentsPerTable + j).getTableNames());
```

Comment:
nit: could use `Collections.singletonList` here instead of `Arrays.asList`

Reply:
Done